Develop

Zborovskii opened the pull request.

Zborovskii committed:
row commit with start front

Zborovskii committed:
fixed other part of code

Zborovskii committed:
add lombok

vadim-gleyf commented on src/main/java/ru/wallet/config/WebSecurityConfig.java:
```diff
@@ -0,0 +1,46 @@
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    UserService userService;
```
модификаторы доступа везде проверь

Zborovskii replied:
проверил, поправил

vadim-gleyf commented on src/main/java/ru/wallet/services/UserService.java:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    private void isValidUser(UserDto user) {
+        User foundUser = userRepository.findByName(user.getName());
```
каждый раз такое себе в базу ходить 

держи в токене все что тебе надо

vadim-gleyf commented on pom.xml:
```diff
@@ -33,38 +45,60 @@
       <artifactId>liquibase-core</artifactId>
     </dependency>
-
+    <dependency>
```
два раза подключаешь

Zborovskii replied:
удалил лишнюю

Zborovskii committed:
add constraint annotations
fixed errors while start app
added catching errors in controllers
fixed info about users's ids
added modificators

Zborovskii committed:
add test
add mapper method
add DTO

vadim-gleyf commented on src/test/java/ru/wallet/config/TestConfig.java:
```diff
@@ -0,0 +1,50 @@
+@EnableJpaRepositories(basePackages = "ru.wallet.repository")
+@EntityScan(basePackages = "ru.wallet.domain")
+@ActiveProfiles({"test"})
+public class TestConfig {
```
зачем этот конфиг?

никакого тест конфига не должно быть

Zborovskii committed:
delete wrong tests config
fixed autoincrement
delete constraint validation

Zborovskii committed:
added access annotation to Wallet controller

Zborovskii committed:
added access annotation to Wallet controller

vadim-gleyf commented on src/main/java/ru/wallet/services/MySecurityService.java:
```diff
@@ -0,0 +1,42 @@
+    public boolean hasWalletOwnerPermission(Object principal, Long walletId) {
+        User user = (User) principal;
+        Optional<Wallet> wallet = walletRepository.findById(walletId);
+        boolean result = false;
```
переделай без флага, сразу return и все в пайплайне делай что надо

vadim-gleyf commented on src/main/java/ru/wallet/services/MySecurityService.java:
```diff
@@ -0,0 +1,42 @@
+    public boolean hasWalletUserPermission(Authentication authentication, Long walletId) {
+        User user = (User) authentication.getPrincipal();
+        Optional<Wallet> wallet = walletRepository.findById(walletId);
+        boolean result = false;
```
тоже самое

vadim-gleyf reviewed (changes requested):
+ тесты не работают, бд в докере должны запускать

Zborovskii committed:
fixed tests container
